### PR TITLE
tdm-gcc: add version 10.3.0

### LINKED ISSUE #237: App Request : TDM-GCC
Can we have [TDM-GCC](https://jmeubank.github.io/tdm-gcc/about/) in the bucket too?
It's a really good alternative to using vanilla gcc.

### PR DESCRIPTION
**Pull request by tech189**
Closes #237

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

File: bucket/tdm-gcc.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        }
+    },
+    "pre_install": [
+        "Expand-7ZipArchive \"$dir\\`$PLUGINSDIR\\*.xz\" \"$dir\"",
+        "Expand-7ZipArchive \"$dir\\`$PLUGINSDIR\\*.lzma\" \"$dir\"",
+        "Expand-7ZipArchive \"$dir\\*.tar\" \"$dir\"",
+        "Get-ChildItem \"$dir\\`$_19_\\*.bat\" | Copy-Item -Destination \"$dir\" -ErrorAction SilentlyContinue",
+        "Get-ChildItem \"$dir\\*.tar\" | Remove-Item -Recurse -Force",
+        "Remove-Item \"$dir\\`$PLUGINSDIR\"-Recurse -Force",
+        "Remove-Item \"$dir\\`$_19_\"-Recurse -Force"
```
Comment (rashil2000) [inline bucket/tdm-gcc.json:23]:
```suggestion
        "Get-ChildItem \"$dir\\`$_19_\\*.bat\" | Copy-Item -Destination \"$dir\" -ErrorAction SilentlyContinue",
        "Copy-Item \"$dir\\bin\\mingw32-make.exe\" \"$dir\\bin\\make.exe\"",
        "Get-ChildItem \"$dir\\*.tar\" | Remove-Item -Recurse -Force",
        "Remove-Item \"$dir\\`$PLUGINSDIR\" -Recurse -Force",
        "Remove-Item \"$dir\\`$_19_\" -Recurse -Force"
```

**Review by rashil2000 — CHANGES_REQUESTED** [review]

**Comment by rashil2000** [comment]
Tested gcc with a simple C program 👍

**Comment by tech189** [comment]
Great, thanks for testing!

**Comment by rashil2000** [comment]
Thanks so much!

**Pull request merged**